inst/report/scripts/run_molevolvr_pipeline.R

```diff
@@ -0,0 +1,1296 @@
+  ## Add row number info to dataframe
+  clust_out[, "RowNum"] <- rows
+  # Name columns
+  colnam <- list("AccNum", "NoAccs", "ClusterID", "NumRows")
+  ## Create cluster name from row number (num of clusters) and num of accessions
+  clust_out <- clust_out %>%
+    `colnames<-`(colnam) %>%
+    mutate(ClusterID = paste0(NumRows, ".", NoAccs))
+
+  # Store data frame column as vector
+  myvar <- c("AccNum", "ClusterID")
+  ## Add only the 2 colummns wanted to a new varible
+  clusters <- clust_out[myvar]
+  # Initialize empty data frame
+  new_clust <- data.frame(ClusterID = character(0),
+                          AccNum = character(0),
+                          stringsAsFactors = F)
+
+  ## Assigning each sseqid to a ClusterID
+  # Iterate over dataframe
+  for (i in 1:nrow(clusters)) {
+    # Extract the cluster name for this row
+    cname <- clusters$ClusterID[i]
+    # Split accession numbers by a space
+    # vals is a vecto of all accession numbers in a row
+    vals <- clusters$AccNum[i] %>%
+      strsplit(split = " ") %>%
+      unlist()
+
+    # Iterate over each element in vals
+    for (v in vals) {
+      # add each accession num w/ corresponding cluster ID to new df
+      new_clust[nrow(new_clust) + 1, ] <- c(cname, v)
+    }
+  }
+  # blast_out$AccNum <- gsub("^>", "", blast_out$AccNum)
+  # blast_out$AccNum <- paste0(">", blast_out$AccNum)
+  blast_clustnames <- merge(blast_out, new_clust, by = "AccNum")
+
+  for (i in 1:nrow(blast_clustnames)) {
+    blast_clustnames$ClusterID[i] <- str_c(blast_clustnames$ClusterID[i])
+  }
+
+  first_prot <- as.data.frame(word(clust_out$AccNum), word(clust_out$ClusterID))
+
+  ## write the new file as a TSV file
+  newarg <- gsub(".bclust.L[0-9][0-9]S[0-9][0-9].tsv", "", clust)
+  # accnum + clusterID
+  write_tsv(new_clust, file = paste0(newarg, ".clustIDs"), append = F)
+  # first protein from every cluster
+  write_tsv(first_prot, file = paste0(newarg, ".clust_reps"),
+            col_names = F, append = F)
+  # cleaned up blast file + clusterIDs
+  write_tsv(blast_clustnames, file = paste0(newarg, ".cln.clust.tsv"),
+            col_names = T, append = F)
+}
+
+# Function to run InterProScan
+run_interproscan <- function(query_file, prefix, outdir) {
+
+  # Start InterProScan run
+  cat("\n######################\n")
+  cat("BEGIN INTERPROSCAN RUN\n")
+  cat("Input file:", query_file, "\n")
+  cat("######################\n")
+
+  # Output file path
+  outfile <- file.path(outdir, paste0(prefix, ".iprscan"))
+
+  # Process the input query file with InterProScan
+  cat("Now processing", query_file, "\n")
+
+  # Run InterProScan command
+  # Construct the command
+  command <- paste(
+    "~/iprdir/interproscan-5.70-102.0/interproscan.sh -i",
+    shQuote(query_file),
+    "-b", shQuote(outfile),
+    "-f TSV --cpu", Sys.getenv("INTERPROSCAN_CPUS", "4"),
+    "--appl Pfam,MobiDBlite,Phobius,Coils,SignalP_GRAM_POSITIVE,",
+    "SignalP_GRAM_NEGATIVE,Hamap,Gene3D,SignalP_EUK"
+  )
```

Comment:
I personally prefer to bring in specific paths to commands from the environment, and failing that assume they're in the user's PATH. With your code as-is, it forces the user to put interproscan in their home directory (well, or to symlink it there) and would require changes whenever we upgraded interproscan, since the version is hardcoded in the path.

I'd personally go with something like the following:

```suggestion
  # Run InterProScan command
  # Construct the command
  
  # get the path to the interproscan.sh script from the environment
  # variable INTERPROSCAN_CMD, or assume it's on the path if unspecified
  iprscan_cmd <- Sys.getenv("INTERPROSCAN_CMD", unset="interproscan.sh")

  command <- paste(
    iprscan_cmd, "-i",
    shQuote(query_file),
    "-b", shQuote(outfile),
    "-f TSV --cpu", Sys.getenv("INTERPROSCAN_CPUS", "4"),
    "--appl Pfam,MobiDBlite,Phobius,Coils,SignalP_GRAM_POSITIVE,",
    "SignalP_GRAM_NEGATIVE,Hamap,Gene3D,SignalP_EUK"
  )
```

Reply:
thanks this was super helpful